### PR TITLE
[Feature](Nereids) support delete using syntax to delete data from unique key table

### PR DESCRIPTION
**Pull request by sohardforaname**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -49,6 +49,10 @@ statement
         SET updateAssignmentSeq
         fromClause?
         whereClause                                                    #update
+    | explain? DELETE FROM tableName=multipartIdentifier tableAlias
+        (PARTITION partition=identifierList)?
+        (USING relation (COMMA relation)*)?
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:54]:
```suggestion
        USING relation (COMMA relation)*
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private final List<String> partitions;
+    private LogicalPlan logicalQuery;
+    private OlapTable targetTable;
+
+    public DeleteCommand(List<String> nameParts, String tableAlias, List<String> partitions, LogicalPlan logicalQuery) {
+        super(PlanType.DELETE_COMMAND);
+        this.nameParts = nameParts;
+        this.tableAlias = tableAlias;
+        this.partitions = partitions;
+        this.logicalQuery = logicalQuery;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/DeleteCommand.java:59]: ImmutableList.copyOf and Objects.requireNonNull

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by Kikyou1997 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java [hunk 3]
```diff
@@ -171,10 +174,6 @@ private void constructInsertStmt() throws AnalysisException {
                 LimitElement.NO_LIMIT
         );
         boolean isPartialUpdate = false;
-        if (((OlapTable) targetTable).getEnableUniqueKeyMergeOnWrite()
-                && cols.size() < targetTable.getColumns().size()) {
-            isPartialUpdate = true;
-        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java:177]: why remove this?

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Pull request merged**